Sources/Moya/TargetType.swift

```diff
@@ -49,14 +50,26 @@ public enum UploadType {
 public enum DownloadType {
 
     /// Download a file to a destination.
-    case request(DownloadDestination)
+    case destination(DownloadDestination)
+
+    /// Download a file to a destination with extra parameters using the given encoding.
+    case encoded(DownloadDestination, parameters: [String: Any], encoding: ParameterEncoding)
 }
 
 /// Represents an HTTP task.
 public enum Task {
 
-    /// A basic request task.
-    case request
+    /// A requests body set with data.
+    case requestData(Data)
+
+    /// A requests body set with parameters and encoding.
+    case requestEncoded(parameters: [String: Any], encoding: ParameterEncoding)
```

Comment:
maybe `request`/`requestParameters`? or `requestWithParameters`? I feel like `encoded` is not as a user-friendly name as `parameters` 😄 `requestWithParameters` reminds me of Swift 2.0 though 😞